src/core/chart-formatters.js

```diff
@@ -0,0 +1,328 @@
+        //add it to the list.
+        categories.push(category);
+        if(categories.length > colourList.length) {
+          throw new Error("Error: too many data categories for colour palette");
+        }
+        index = categories.length - 1;
+      }
+      colors[seriesName] = colourList[index];
+    }
+  });
+  graph.data.colors = colors;
+  return graph;
+};
+
+/*
+ * Post-processing graph function that visually de-emphasizes certain
+ * data series by lightening their assigned colour. (Assumes the graph
+ * has a white background, otherwise lightening isn't de-emphasizing.)
+ *
+ * Accepts a C3 graph object and a ranking function. The ranking function
+ * will be applied to each data column in the graph object, and should
+ * output a number between 0 and 1, which will be used to determine the
+ * visual prominence of the associated data series. Series ranked 1 will
+ * be drawn normally with their assigned colour, values less than one and
+ * greater than zero will be lightened proportionately. A data series ranked
+ * 0 by the ranking function will be drawn in white.
+ *
+ * Returns the graph object, modified by the addition of a data.color
+ * function to operate on assigned series colours.
+ * Each data column passed to the ranking function is an array like this:
+ *
+ * ['Monthly Mean Tasmin', 30, 20, 50, 40, 60, 50, 10, 10, 20, 30, 40, 50]
+ */
+function fadeSeriesByRank (graph, ranker) {
+
+  var rankDictionary = {};
+
+  _.each(graph.data.columns, column => {
+    var seriesName = column[0];
+    if(!_.isEqual(seriesName, "x")) {
+      rankDictionary[seriesName] = ranker(column);
+    }
+  });
+
+  //c3 will pass the function the assigned colour, and either:
+  //     * a string with the name of the time series (drawing legend)
+  //     * an object with attributes about the time series (drawing a point or line)
+  var fader = function(colour, d) {
+    var scale = chroma.scale(['white', colour]);
+    if(_.isObject(d)) { //d = data attributes
+      return scale(rankDictionary[d.id]).hex();
+    }
+    else { //d = series name only
+      return scale(rankDictionary[d]).hex();
+    }
+  };
+
+  graph.data.color = fader;
+  return graph;
+};
+
+/*
+ * Post-processing graph function that removes data series from the legend.
+ *
+ * Accepts a C3 graph and a predicate function. Applies the predicate to
+ * each data series. If the predicate returns true, the data series will
+ * be hidden from the legend. If the predicate returns false, the data series
+ * will appear in the legend as normal.
+ *
+ * By default, every data series appears in the legend; this postprocessor
+ * is only needed if at least one series should be hidden.
+ */
+function hideSeriesInLegend (graph, predicate) {
+  var hiddenSeries = [];
+
+  _.each(graph.data.columns, column => {
+    var seriesName = column[0];
+    if(!_.isEqual(seriesName, "x")) {
+      if(predicate(column)){
+        hiddenSeries.push(seriesName);
+      }
+    }
```

Comment:
Ditto comments re. `_.each` vs. for-of, and `!==` vs. `isEqual` for simple values.
Also, the nested if statements should be collapsed to one.

FWIW, this could be written as
```
const hiddenSeries = 
   _.chain(graph.data.columns)
   .filter(column => column[0] !== 'x' && predicate(column))
   .map(column => column[0]);
```
which depending on reader might be deemed easier to read.